src/mqt/benchviewer/backend.py

```diff
@@ -593,73 +656,74 @@ def get_target_device(filename: str):
     raise ValueError("Unknown target device: " + filename)
 
 
-def get_compiler_and_settings(filename: str):
+def get_compiler_and_settings(filename: str) -> tuple[str, str | int | None]:
     if "qiskit" in filename:
         return "qiskit", get_opt_level(filename)
     if "tket" in filename:
         return "tket", get_tket_settings(filename)
     raise ValueError("Unknown compiler: " + filename)
 
 
-def parse_data(filename: str):
-    """Extracts the necessary information from a given filename.
-
+def create_database(zip_file: ZipFile) -> pd.DataFrame:
+    """Creates the database based on the provided directories.
     Keyword arguments:
-    filename -- name of file
-
+    qasm_path -- zip containing all .qasm files
     Return values:
-    parsed_data -- parsed data extracted from filename
+    database -- database containing all available benchmarks
     """
-    benchmark = filename.split("_")[0].lower()
-    num_qubits = get_num_qubits(filename)
-    indep_flag = "indep" in filename
-    nativegates_flag = "nativegates" in filename
-    mapped_flag = "mapped" in filename
-    compiler, compiler_settings = get_compiler_and_settings(filename)
-    gate_set = get_gate_set(filename) if nativegates_flag or mapped_flag else None
-    target_device = get_target_device(filename) if mapped_flag else None
-
-    return [
-        benchmark,
-        num_qubits,
-        indep_flag,
-        nativegates_flag,
-        mapped_flag,
-        compiler,
-        compiler_settings,
-        gate_set,
-        target_device,
-        filename,
-    ]
-
-
-def get_opt_level(filename: str):
-    """Extracts the optimization level based on a filename.
-
-    Keyword arguments:
-    filename -- filename of a benchmark
-
-    Return values:
-    num -- optimization level
-    """
-
-    pat = re.compile(r"opt\d")
-    m = pat.search(filename)
-    num = m.group()[-1:] if m else -1
-    return int(num)
-
-
-def get_num_qubits(filename: str):
-    """Extracts the number of qubits based on a filename.
-
-    Keyword arguments:
-    filename -- filename of a benchmark
-
-    Return values:
-    num -- number of qubits
-    """
-
-    pat = re.compile(r"(\d+)\.")
-    m = pat.search(filename)
-    num = m.group()[0:-1] if m else -1
-    return int(num)
+    rows_list = []
+
+    for filename in zip_file.namelist():
+        if filename.endswith(".qasm"):
+            parsed_data = parse_data(filename)
+            rows_list.append(parsed_data)
+
+    colnames = list(ParsedBenchmarkName.__annotations__.keys())
+
+    database = pd.DataFrame(rows_list, columns=colnames)
+    database["num_qubits"] = database["num_qubits"].astype(int)
+    database["benchmark"] = database["benchmark"].astype(str)
+    return database
+
+
+def handle_downloading_benchmarks(target_location: str, download_url: str) -> None:
```

Comment:
shouldn't that function be covered by tests?

Reply:
Actually, a mistake happened here since this method is twice in the file: Once as a class method and once without. I removed the non-class method. The class method was covered by tests.